lib/puppet/provider/cisco_snmp_user/cisco.rb

```diff
@@ -214,7 +214,7 @@ def handle_attribute_update
   end
 
   def unconfigure_snmp_user
-    @snmp_user.destroy
+    @snmp_user.destroy unless @resource[:user] == 'admin'
```

Comment:
Please add a comment for future developers to indicate why we don't call destroy for the `admin` user.